pkg/sdkserver/sdkserver.go

```diff
@@ -409,7 +430,13 @@ func (s *SDKServer) gameServer() (*agonesv1.GameServer, error) {
 	// they will block here until it's ready
 	s.gsWaitForSync.Wait()
 	gs, err := s.gameServerLister.GameServers(s.namespace).Get(s.gameServerName)
-	return gs, errors.Wrapf(err, "could not retrieve GameServer %s/%s", s.namespace, s.gameServerName)
+	if err != nil {
+		return gs, errors.Wrapf(err, "could not retrieve GameServer %s/%s", s.namespace, s.gameServerName)
+	}
+	if s.gsCopy != nil && gs.ObjectMeta.Generation < s.gsCopy.Generation {
```

Comment:
We probably need a read lock here as well, yeah? Since we're reading `s.gsCopy`

Reply:
Oh yep good point, I'll add that in.